lib/util/stat.js

```diff
@@ -147,12 +147,35 @@ function checkParentPathsSync (src, srcStat, dest, funcName) {
     if (err.code === 'ENOENT') return
     throw err
   }
-  if (destStat.ino && destStat.dev && destStat.ino === srcStat.ino && destStat.dev === srcStat.dev) {
+  if (areApparentlyIdentical(srcStat, destStat)) {
     throw new Error(errMsg(src, dest, funcName))
   }
   return checkParentPathsSync(src, srcStat, destParent, funcName)
 }
 
+function areApparentlyIdentical (srcStat, destStat) {
```

Comment:
This is good but the function name is a bit confusing. Will you please rename it to something simpler like just `areIdentical`?

Reply:
Sure thing

Reply:
Fixed in https://github.com/jprichardson/node-fs-extra/pull/694/commits/7e4c8525ea031d89cd8e7d40245afc3b753c7559